fuel-vm/src/checked_transaction.rs

```diff
@@ -1383,179 +1416,237 @@ mod tests {
     #[test]
-    fn bytes_fee_cant_overflow() {
+    fn into_immutable__bytes_fee_cant_overflow() {
         let rng = &mut StdRng::seed_from_u64(2322u64);
 
         let input_amount = 1000;
-        let gas_price = Word::MAX;
+        let max_gas_price = Word::MAX;
         let gas_limit = 0; // ensure only bytes are included in fee
-        let transaction = base_asset_tx(rng, input_amount, gas_limit);
+        let zero_fee_limit = 0;
+        let transaction = base_asset_tx(rng, input_amount, gas_limit, zero_fee_limit);
+        let gas_costs = GasCosts::default();
 
         let consensus_params = params(1);
 
+        let fee_params = consensus_params.fee_params();
         let err = transaction
-            .into_checked(Default::default(), &consensus_params, gas_price)
+            .into_checked(Default::default(), &consensus_params)
+            .unwrap()
+            .into_ready(max_gas_price, &gas_costs, fee_params)
             .expect_err("overflow expected");
 
         assert_eq!(err, CheckError::Validity(ValidityError::BalanceOverflow));
     }
 
-    fn arb_tx(rng: &mut StdRng) -> Script {
-        let input_amount = 1000;
-        let gas_limit = 1000;
-        base_asset_tx(rng, input_amount, gas_limit)
-    }
-
     #[test]
-    fn into_checked_basic__min_fee_calc_includes_tip() {
+    fn into_immutable__fails_if_fee_limit_too_low() {
         let rng = &mut StdRng::seed_from_u64(2322u64);
-        let gas_price = 1;
-        let mut tx = arb_tx(rng);
 
-        // given
-        let tipless_tx = tx.clone();
+        let input_amount = 1000;
+        let gas_price = 100;
+        let gas_limit = 0; // ensure only bytes are included in fee
+        let gas_costs = GasCosts::default();
 
-        let min_fee_without_tip = tipless_tx
-            .into_checked_basic(1.into(), &ConsensusParameters::standard(), gas_price)
-            .unwrap()
-            .metadata()
-            .fee
-            .min_fee();
+        let consensus_params = params(1);
 
-        let tip = 100;
+        let fee_params = consensus_params.fee_params();
 
-        // when
-        tx.set_tip(tip);
+        // given
+        let zero_fee_limit = 0;
+        let transaction = base_asset_tx(rng, input_amount, gas_limit, zero_fee_limit);
 
-        let min_fee_with_tip = tx
-            .into_checked_basic(1.into(), &ConsensusParameters::standard(), gas_price)
+        // when
+        let err = transaction
+            .into_checked(Default::default(), &consensus_params)
             .unwrap()
-            .metadata()
-            .fee
-            .min_fee();
+            .into_ready(gas_price, &gas_costs, fee_params)
+            .expect_err("overflow expected");
 
         // then
-        assert_eq!(min_fee_without_tip + tip, min_fee_with_tip);
+        assert!(matches!(err, CheckError::InsufficientMaxFee { .. }));
     }
 
     #[test]
-    fn into_checked_basic__max_fee_calc_includes_tip() {
+    fn into_immutable__tx_fails_if_tip_not_covered() {
```

Comment:
Maybe we want to change the name of all test from `into_immutable_...` -> `into_ready_...`